eZ/Publish/SPI/FieldType/Comparable.php

```diff
@@ -0,0 +1,12 @@
+<?php
+
+/**
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+namespace eZ\Publish\SPI\FieldType;
+
+interface Comparable
+{
+    public function isEquals($newHash, $currentHash): bool;
```

Comment:
This should not compare hashes but VO.
```
function isEquals(value) 
    return value === self
```
you could transform value into hash in that method and compare those in abstract implementation.